### PR TITLE
feat: sam remote invoke help text and UX fixes

### PR DESCRIPTION
**Pull request by hnnasit**
#### Which issue(s) does this change fix?
N/A
Changes:
- Fixes UX when `NoRegionError` and `NoCredentialsError` errors are thrown by wrapping command with `command_exception_handler`.
- Fixes UX for error when both `--event` and `--event-file` are passed
- Improve help text for `sam remote invoke`
- Change `output-format` option to `output` with values `text/json`
- Allow remote invoke when an empty event is passed for lambda
- Read from stdin when `-` is passed for `event-file`
- Log the function name before invoking the resource

After:
modified_remote_invoke_help_text

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by mndeveci — APPROVED** [review]

File: samcli/commands/remote/invoke/core/command.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            formatter.write_rd(
+                [
+                    RowDefinition(
+                        name="ARN",
+                        text="Amazon Resource Name",
+                        extra_row_modifiers=[ShowcaseRowModifier()],
+                    ),
+                    RowDefinition(
+                        name="CFN",
+                        text="CloudFormation",
```
Comment (sriram-mv) [inline samcli/commands/remote/invoke/core/command.py:131]: AWS Cloudformation, do we reference it anywhere else?

File: samcli/commands/_utils/options.py [hunk 2]
```diff
@@ -594,7 +594,13 @@ def remote_invoke_parameter_click_option():
         type=RemoteInvokeBotoApiParameterType(),
         callback=remote_invoke_boto_parameter_callback,
         required=False,
-        help="Additional parameters for the boto API call.\n" "Lambda APIs: invoke and invoke_with_response_stream",
+        help="Additional parameters that can be passed to the boto API call to invoke the resource.\n"
```
Comment (sriram-mv) [inline samcli/commands/_utils/options.py:597]: the customer should not know about the implementation details such as `boto`. we should remove that.

File: samcli/commands/remote/invoke/cli.py [hunk 3]
```diff
@@ -25,25 +27,40 @@
 """
 SHORT_HELP = "Invoke a deployed resource in the cloud"
 
+DESCRIPTION = """
+  Invoke or send an event to cloud resources.
```
Comment (sriram-mv) [inline samcli/commands/remote/invoke/cli.py:31]: grammar

File: samcli/commands/remote/invoke/core/command.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                            name=style(
+                                f"$ echo {json.dumps({'message':'hello!'})} | "
+                                f"{ctx.command_path} HelloWorldFunction -e <>"
+                            ),
+                            extra_row_modifiers=[ShowcaseRowModifier()],
+                        ),
+                    ]
+                )
+            with formatter.indented_section(
+                name="Invoke lambda function using lambda ARN and get full boto response", extra_indents=1
```
Comment (sriram-mv) [inline samcli/commands/remote/invoke/core/command.py:68]: get "AWS API response" instead?

File: samcli/commands/remote/invoke/core/options.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+"""
+from typing import Dict, List
+
+from samcli.cli.core.options import ALL_COMMON_OPTIONS, add_common_options_info
+from samcli.cli.row_modifiers import RowDefinition
+
+# NOTE: The ordering of the option lists matter, they are the order
+# in which options will be displayed.
+
+INFRASTRUCTURE_OPTION_NAMES: List[str] = ["stack_name"]
```
[the remark this answers is not in the document]
Comment (sriram-mv) [reply samcli/commands/remote/invoke/core/options.py:12]: We can consolidate for sure!

**Review by sriram-mv — CHANGES_REQUESTED** [review]

File: samcli/commands/remote/invoke/core/command.py (hunk 1 above)
Comment (hnnasit) [reply samcli/commands/remote/invoke/core/command.py:131]: We reference it in the help text for the command: `Invoke or send an event to cloud resources in your CFN stack`

Comment (hnnasit) [reply samcli/commands/remote/invoke/core/command.py:131]: Actually, I will remove it as the command is not restricted to CFN stack.

**Review by mndeveci — APPROVED** [review]

File: samcli/lib/cli_validation/remote_invoke_options_validations.py [hunk 6]
```diff
@@ -48,10 +46,9 @@ def wrapped(*args, **kwargs):
 
         validator.validate()
 
-        # if no event nor event_file arguments are given, read from stdin
-        if not event and not event_file:
-            LOG.debug("Neither --event nor --event-file options have been provided, reading from stdin")
-            kwargs["event_file"] = cast(TextIOWrapper, sys.stdin)
+        # If "-" is provided for --event-file, click uses it as a special file to refer to stdin.
+        if event_file and event_file.fileno() == sys.stdin.fileno():
```
Comment (sriram-mv) [inline samcli/lib/cli_validation/remote_invoke_options_validations.py:50]: just checking: is this the same behavior as local?

**Review by sriram-mv — APPROVED** [review]

**Review by mildaniel — APPROVED** [review]

File: samcli/lib/cli_validation/remote_invoke_options_validations.py (hunk 6 above)
Comment (hnnasit) [reply samcli/lib/cli_validation/remote_invoke_options_validations.py:50]: Yes, `sam local` behaves in a similar way when `-` is passed to `-e` file path option. Click intelligently treats this as being read from `stdin` as the file stream.

**Pull request merged**